python/sparkdl/transformers/tf_tensor.py

```diff
@@ -0,0 +1,105 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import logging
+import numpy as np
+import tensorflow as tf
+import tensorframes as tfs
+
+from pyspark.ml import Transformer
+from pyspark.ml.param import Param, Params
+from pyspark.sql.functions import udf
+
+from sparkdl.graph.builder import IsolatedSession
+import sparkdl.graph.utils as tfx
+from sparkdl.transformers.param import (
+    keyword_only, HasInputCol, HasOutputCol, SparkDLTypeConverters,
+    HasTFGraph, HasInputTensor, HasOutputTensor)
+
+__all__ = ['TFOneDimTensorTransformer']
+
+logger = logging.getLogger('sparkdl')
+    
+def _tranform_with_graph_function(gfn, df, input_col, output_col):
+    """ Apply a transform with a graph function
+    """
+    analyzed_df = tfs.analyze(df)
+
+    with IsolatedSession() as issn:
+        feeds, fetches = issn.importGraphFunction(gfn, prefix='')
+        assert len(feeds) == 1, 'only support single input TF model'
+        assert len(fetches) == 1, 'only support single output TF model'
+
+        # Change the column name for TensorFrames
+        orig_in_name = tfx.op_name(issn.graph, feeds[0])
+        input_df = analyzed_df.withColumnRenamed(input_col, orig_in_name)
+
+        output_df = tfs.map_blocks(fetches, input_df)
+
+        # Change the names back
+        orig_out_name = tfx.op_name(issn.graph, fetches[0])
+        final_df = (output_df
+                    .withColumnRenamed(orig_out_name, output_col)
+                    .withColumnRenamed(orig_in_name, input_col))
+
+    return final_df
+
+class TFOneDimTensorTransformer(Transformer, HasInputCol, HasOutputCol,
+                                HasTFGraph, HasInputTensor, HasOutputTensor):
+    """
+    Applies the TensorFlow graph to the array column in DataFrame.
+
+    Restrictions of the current API:
+
+    We assume that
+    - All graphs have a "minibatch" dimension (i.e. an unknown leading
+      dimension) in the tensor shapes.
+    - Input DataFrame has an array column where all elements have the same length    
+    """
+    
+    @keyword_only
+    def __init__(self, tfGraph=None, inputTensor=None, outputTensor=None,
+                 inputCol=None, outputCol=None):
+        """
+        __init__(self, tfGraph=None, inputTensor=None, outputTensor=None,
+                 inputCol=None, outputCol=None)
+        """
+        super(TFOneDimTensorTransformer, self).__init__()
+        kwargs = self._input_kwargs
+        self.setParams(**kwargs)
+
+    @keyword_only
+    def setParams(self, tfGraph=None, inputTensor=None, outputTensor=None,
+                  inputCol=None, outputCol=None):
```

Comment:
I am wondering if we want the tensors to be in separate columns or in fields of a nested column?